src/Servers/Kestrel/Core/test/Http2StreamStackTests.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests
+{
+    public class Http2StreamStackTests
+    {
+        [Fact]
+        public void RemoveExpired_Empty_NoOp()
+        {
+            var streams = new Http2StreamStack(10);
+
+            streams.RemoveExpired(100);
+        }
+
+        [Fact]
+        public void RemoveExpired_NoneExpired_NoOp()
+        {
+            var streams = new Http2StreamStack(10);
+            streams.Push(CreateStream(streamId: 1, expirationTicks: 200));
+
+            streams.RemoveExpired(100);
+
+            Assert.Equal(1, streams.Count);
+            Assert.Equal(1, ((Http2Stream)streams._array[0]).StreamId);
+        }
+
+        [Fact]
+        public void RemoveExpired_OneExpired_ExpiredStreamRemoved()
```

Comment:
It'd be nice to have at least one test that verifies this can remove more than one stream at a time.